dropgz/build/cni.Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM mcr.microsoft.com/oss/go/microsoft/golang:1.18 AS builder
+ARG VERSION
+WORKDIR /azure-container-networking
+COPY . .
+RUN CGO_ENABLED=0 go build -a -o bin/azure-vnet -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/network/plugin/main.go
```

Comment:
could you help me understand the steps required to add CNI binaries from another image? Particularly interested in how this interacts with multi-arch builds (I'm not very familiar with that process)

Reply:
Answering your second question first: we don't need to worry about multi-arch here, that is handled automatically through [flags[1]](https://github.com/rbtr/azure-container-networking/blob/f09087817e0d802c34ea1a8a46f3f107a99164be/.pipelines/pipeline.yaml#L161-L168) [passed[2]](https://github.com/rbtr/azure-container-networking/blob/f09087817e0d802c34ea1a8a46f3f107a99164be/.pipelines/containers/container-template.yaml#L23-L24) [to[3]](https://github.com/rbtr/azure-container-networking/blob/f09087817e0d802c34ea1a8a46f3f107a99164be/Makefile#L248-L254) the [container builder[4]](https://github.com/rbtr/azure-container-networking/blob/f09087817e0d802c34ea1a8a46f3f107a99164be/Makefile#L181-L188). Any `FROM <image>` in this Dockerfile will do the right thing as long as `<image>` is also a multiplat image.
Which means, to your first question, we can leverage multi-stage builds and reference another CNI image like `FROM docker.io/other/cni:x.x.x as othercni`.  Then we add a line to copy bins out of that image into our target dir, right before we calculate checksums and compress (which is the next line in this file): 
```Dockerfile
FROM docker.io/other/cni:x.x.x as othercni

[do other prep]
COPY --from=othercni /bin/othercnibin /dropgz/pkg/embed/gz/othercnibin
[checksum and compress]
...
```